examples/service_worker/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+//! A simple example demonstrating how to integrate service worker into Seed.
+//! This example will cover the following:
+//! 1. Cache resources
+//! 2. Register the service worker
+//! 3. If the service worker is not yet activated, an even listener will be registered, waiting for the
+//!    state to reach "activated".
+//! 4. When the state reaches "activated", the Notification object will request permission for notifications
+//! 5. If permission is granted, the `PushManager` will subscribe using an example vapid key
+//! 6. Finally, a `PushSubscription` will be returned, containing the information that can be passed to a
+//!    notification back-end server.
+
+#![allow(clippy::cast_possible_truncation, clippy::needless_pass_by_value)]
```

Comment:
I changed to the following for now:
```
#![allow(clippy::needless_pass_by_value)]
```

If you're ok with the clippy error (I think it will break the build?) let me know and I'll take it out completely.

Reply:
`#![allow(clippy::needless_pass_by_value)]` is ok. 
I don't know how to eliminate it, it looks like the problem is hidden in generated code by wasm-bindgen